crates/codegen/runtime/generator/src/parser/codegen/scanner_definition.rs

```diff
@@ -1,132 +1,216 @@
 use std::collections::BTreeSet;
 
+use codegen_language_definition::model::{self, Identifier};
 use inflector::Inflector;
 use proc_macro2::TokenStream;
 use quote::{format_ident, quote};
 
 use crate::parser::codegen::versioned::VersionedQuote;
-use crate::parser::grammar::{ScannerDefinitionNode, ScannerDefinitionRef};
+use crate::parser::grammar::ScannerDefinition;
 
-pub trait ScannerDefinitionCodegen {
```

Comment:
Name, again. I won't repeat this comment further.

Reply:
See https://github.com/NomicFoundation/slang/pull/1003#discussion_r1641963402 (repeated for visibility)